scipy/spatial/distance.py

```diff
@@ -2190,6 +2217,7 @@ def pdist(X, metric='euclidean', *, out=None, **kwargs):
 
         if metric_info is not None:
             pdist_fn = metric_info.pdist_func
+            _extra_windows_error_checks(X, out, "pdist", m=m, **kwargs)
```

Comment:
I had mean to pass `required_shape` as an argument, but it's not a big deal
```suggestion
            _extra_windows_error_checks(X, out, required_shape=(m * (m - 1) / 2,), **kwargs)
```